pkg/volume/vsphere_volume/vsphere_volume.go

```diff
@@ -384,7 +384,7 @@ func (v *vsphereVolumeProvisioner) Provision(selectedNode *v1.Node, allowedTopol
 	pv := &v1.PersistentVolume{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:   v.options.PVName,
-			Labels: map[string]string{},
+			Labels: volSpec.Labels,
```

Comment:
The PVL admission controller handles labelling/node affinity rules for manually created volumes. But for dynamically created volumes, we still have to apply the volume's node affinity rules here as part of `Provision()`.

See [GCE PD](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/gcepd/gce_pd.go#L539-L565) and [AWS EBS](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/awsebs/aws_ebs.go#L573-L588) for examples.

Reply:
The PVL admission control seems to be handling the nodeAffinity for dynamic volumes as well. Still I will add this logic to the Provision() method in vsphere_volume just to be consistent with other volume plugins. The latest commit 9d6eb69 has this addressed. @andrewsykim PTAL.